core/src/data/value/binary_op/integer/i16.rs

```diff
@@ -0,0 +1,41 @@
+use {
+    crate::{impl_try_binary_op, prelude::Value},
+    rust_decimal::prelude::Decimal,
+    std::cmp::Ordering,
+};
+
+impl_try_binary_op!(I16, i16);
+#[cfg(test)]
+crate::generate_binary_op_tests!(I16, i16);
+
+impl PartialEq<Value> for i16 {
+    fn eq(&self, other: &Value) -> bool {
+        match other {
+            I8(other) => (*self as i8) == *other,
+            I16(other) => self == other,
+            I32(other) => (*self as i32) == *other,
+            I64(other) => (*self as i64) == *other,
+            I128(other) => (*self as i128) == *other,
+            U8(other) => (*self as i64) == (*other as i64),
```

Comment:
```suggestion
            U8(other) => *self == (*other as i16),
```
There doesn't seem to be any need to convert to i64. because it's a comparison